### PR TITLE
shin/ch2084/implement-dashboard-no-delegate-page

### PR DESCRIPTION
**Pull request by sarveshwar-singh**

### CODE REVIEW

**Comment by d-novak** [comment]
Rebased nicarq/connect-delegation-pages

File: app/components/wallet/staking/dashboard/StakingDashboard.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import EpochProgress from './EpochProgress';
+import UserSummary from './UserSummary';
+import GraphWrapper from './GraphWrapper';
+import StakePool from './StakePool';
+import RewardPopup from './RewardPopup';
+import EmptyDashboard from './EmptyDashboard';
+import styles from './StakingDashboard.scss';
+
+type Props = {|
+  totalAdaSum: any,
```
Comment (SebastienGllmt) [inline app/components/wallet/staking/dashboard/StakingDashboard.js]: try and avoid `any` type when possible

File: app/components/wallet/staking/dashboard/GraphWrapper.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.wrapper {
+    display: block;
+    height: 325px; // Dev-visual
```
[the remark this answers is not in the document]
Comment (d-novak) [reply app/components/wallet/staking/dashboard/GraphWrapper.scss]: `Dev-visual` comments are comments I use for stuff that's dev only and will be refactored (ie. some component that is not implemented yet but I want it to have specific dimensions for the layout development).

File: app/components/wallet/staking/dashboard/ProgressCircle.scss [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    font-family: var(--font-medium);
+    font-size: 18px;
+    line-height: 22px;
+    color: #676970;
+    text-align: center;
+    &.bold {
+        font-family: var(--font-bold);
+        font-size: 24px;
+        line-height: 26px;
+        color: #8791AD;
```
[the remark this answers is not in the document]
Comment (d-novak) [reply app/components/wallet/staking/dashboard/ProgressCircle.scss]: Sure, all the colors that are repeated will get refactored to vars, I prefer to do this all at once when I know what colors are actually used, same goes for other properties.

File: package.json [hunk 4]
```diff
@@ -155,6 +155,7 @@
     "crypto-random-string": "3.0.1",
     "es6-error": "4.1.1",
     "file-saver": "1.3.8",
+    "jdenticon": "^2.2.0",
```
Comment (SebastienGllmt) [inline package.json]:
```suggestion
    "jdenticon": "2.2.0",
```

File: app/components/wallet/staking/dashboard/GraphWrapper.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+const GraphTabs = ({ tabs }) => {
+  return (
+    <ul className={styles.tabsWrapper}>
+      {
+        tabs.map(
+          (tab, i) => (
+            <li
+              key={tab}
+              // Change to real isActive? condition for current tab
```
Comment (SebastienGllmt) [inline app/components/wallet/staking/dashboard/GraphWrapper.js]: TODO

File: app/components/wallet/staking/dashboard/EpochProgress.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            <ProgressCircle percentage={percentage} variant="epoch" />
+          </div>
+          <div className={styles.stats}>
+            <div>
+              <p className={classnames(styles.label, styles.dashed)}>Epoch:</p>
+              <p className={styles.value}>5</p>
+            </div>
+            {endTime && (
+              <div className={styles.row}>
+                <p className={styles.label}>Time until Epoch end:</p>
```
Comment (SebastienGllmt) [inline app/components/wallet/staking/dashboard/EpochProgress.js]: hard-coded string

File: app/components/wallet/staking/dashboard/EpochProgress.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+    return (
+      <Card title={intl.formatMessage(messages.title)}>
+        <div className={styles.wrapper}>
+          <div className={styles.chart}>
+            <ProgressCircle percentage={percentage} variant="epoch" />
+          </div>
+          <div className={styles.stats}>
+            <div>
+              <p className={classnames(styles.label, styles.dashed)}>Epoch:</p>
```
Comment (SebastienGllmt) [inline app/components/wallet/staking/dashboard/EpochProgress.js]: hard-coded string

File: app/components/wallet/staking/dashboard/Address.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  hash: string,
+};
+
+@observer
+export default class Address extends Component<Props> {
+
+  render() {
+    const { hash } = this.props;
+
+    // Check if this truncation is okay
```
Comment (SebastienGllmt) [inline app/components/wallet/staking/dashboard/Address.js]: remove this todo

**Pull request merged**